src/renderer/features/wallet-details/model/wallet-proxies-model.ts

```diff
@@ -65,7 +72,16 @@ const { $: $walletProxies } = createStoreFromEffect<
       });
 
       const accountsProxies = await Promise.all(accountProxiesPromises);
-      return [chainId, accountsProxies.flat().filter(nonNullable)];
+      const allProxies = accountsProxies.flat().filter(nonNullable);
+      const uniqueProxies = allProxies.reduce((acc, proxy) => {
+        const key = `${proxy.accountId}_${proxy.proxiedAccountId}_${proxy.proxyType}`;
+        if (!acc.has(key)) {
+          acc.set(key, proxy);
+        }
+        return acc;
+      }, new Map<string, Proxy>());
+
+      return [chainId, Array.from(uniqueProxies.values())];
```

Comment:
I suggest to extract this to a separate function to make this store easier to read as it is rather big now

Reply:
Exctracted into `deduplicateProxies`